From e660cd02a3fd80fe37dfa6264b2d8f3eed1d7c08 Mon Sep 17 00:00:00 2001
From: Aaron Conole <aconole@redhat.com>
Date: Mon, 17 Apr 2017 13:59:50 -0400
Subject: [PATCH 1/3] lib/automake.mk: don't install runtime directories

The Open vSwitch run, log, and DB directories are installed as part of the
normal `make install` process.  However, this means they are created with
user and group ownership that may conflict with the desired user.  For
example, running `make install` as root will install those files as
root:root, whereas the runtime user desired may be openvswitch:openvswitch.

Since these directories are automatically created as part of the ovs-ctl
command, and with the correct user:group permissions, it makes sense to
delay creation until these directories are actually required.

Signed-off-by: Aaron Conole <aconole@redhat.com>
Signed-off-by: Ben Pfaff <blp@ovn.org>
(cherry picked from commit d0c961a99f570ec1973bf5540ba9237b6720c848)
---
 lib/automake.mk | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/lib/automake.mk b/lib/automake.mk
index e4cfc976f..1a9f1f99d 100644
--- a/lib/automake.mk
+++ b/lib/automake.mk
@@ -526,8 +526,5 @@ EXTRA_DIST += build-aux/extract-ofp-msgs
 
 INSTALL_DATA_LOCAL += lib-install-data-local
 lib-install-data-local:
-	$(MKDIR_P) $(DESTDIR)$(RUNDIR)
 	$(MKDIR_P) $(DESTDIR)$(PKIDIR)
-	$(MKDIR_P) $(DESTDIR)$(LOGDIR)
-	$(MKDIR_P) $(DESTDIR)$(DBDIR)
 	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/openvswitch
-- 
2.13.3

